### PR TITLE
feat: move docs to github pages

### LINKED ISSUE #3314: [doc] : missing OnAudioFocusChanged doc
# Bug

OnAudioFocusChanged is missing in documentation

## Platform
any

## Environment info

Library version: alpha.8

## Steps To Reproduce
open doc ...

### PR DESCRIPTION
**Pull request by KrzysztofMoch**
## Summary
Moved docs to Github pages using nextra

fixes:  #3314

fixes: #

## How to test ?
Check docs README to check how to run docs locally

### CODE REVIEW

File: docs/next-env.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/// <reference types="next" />
```
Comment (freeboub) [inline docs/next-env.d.ts:1]: Do we need this file?

**Comment by KrzysztofMoch** [comment]
@freeboub please test it locally and see if it's okay. To publish it on GH pages we will use workflows that i created in this PR (there is no way to test workflows so we will probably need to create new PR for it)

File: docs/pages/component/_meta.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "props": "Configurable props",
```
Comment (freeboub) [inline docs/pages/component/_meta.json]:
```suggestion
  "props": "Properties",
```

File: docs/pages/component/events.md [hunk 3]
```diff
@@ -0,0 +1,469 @@
+# Events
+
```
Comment (freeboub) [inline docs/pages/component/events.md:3]:
```suggestion
This page shows the list of available callbacks to handle player notifications
```

File: docs/pages/component/props.md [hunk 4]
```diff
@@ -0,0 +1,793 @@
+# Configurable props
```
Comment (freeboub) [inline docs/pages/component/props.md:1]:
```suggestion
# Configurable props

This page shows the list of available properties to configure player
```

File: docs/pages/updating.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+#### **android/gradle.properties**
+
+Migrating to AndroidX (needs version >= 5.0.0):
+
+```groovy
+android.useAndroidX=true
+android.enableJetifier=true
+```
+
+### Version 4.0.0
```
Comment (freeboub) [inline docs/pages/updating.md]: don't you think we can remove this update part ?
I agree to remove it

File: docs/pages/_meta.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
+  "index": "Introduction",
+  "installation": "Installation",
+  "component": "Component",
```
Comment (freeboub) [inline docs/pages/_meta.json]: rename component to API ?

File: docs/pages/other/drm.md [hunk 7]
```diff
@@ -31,7 +31,10 @@ Platforms: iOS
 
 ### `getLicense`
 
-`licenseServer` and `headers` will be ignored. You will obtain as argument the `SPC` (as ASCII string, you will probably need to convert it to base 64) obtained from your `contentId` + the provided certificate via `[loadingRequest streamingContentKeyRequestDataForApp:certificateData contentIdentifier:contentIdData options:nil error:&spcError];`.
```
Comment (freeboub) [inline docs/pages/other/drm.md]: can you move DRM inside component part please ?

**Review by freeboub — CHANGES_REQUESTED** [review]
some minor / cosmetic change requested.
Additionally it seems that Method and Static Method sections were lost 
It would need a specifc section I think

### Methods
| Name |Platforms Support  | 
|--|--|
|[dismissFullscreenPlayer](#dismissfullscreenplayer)|Android, iOS|
|[presentFullscreenPlayer](#presentfullscreenplayer)|Android, iOS|
|[save](#save)|iOS|
|[restoreUserInterfaceForPictureInPictureStop](#restoreuserinterfaceforpictureinpicturestop)|iOS|
|[seek](#seek)|All|

### Static methods

| Name |Platforms Support  |
|--|--|
|[getWidevineLevel](#getwidevinelevel)|Android|
|[isCodecSupported](#iscodecsupported)|Android|
|[isHEVCSupported](#ishevcsupported)|Android|

**Comment by freeboub** [comment]
OnAudioFocusChanged not documentated (but it is not your fault :) )

File: docs/next-env.d.ts (hunk 1 above)
Comment (KrzysztofMoch) [reply docs/next-env.d.ts:1]: That was in template, so i would leave it as it is

File: docs/pages/other/drm.md (hunk 7 above)
Comment (KrzysztofMoch) [reply docs/pages/other/drm.md]: ~~I would like to leave it here. We can add link to component part WDYT ?~~
 okay I misunderstood that - will move this

File: docs/pages/_meta.json (hunk 6 above)
Comment (KrzysztofMoch) [reply docs/pages/_meta.json]: Done ✅

File: docs/pages/updating.md (hunk 5 above)
Comment (KrzysztofMoch) [reply docs/pages/updating.md]: Done ✅

File: docs/pages/component/props.md (hunk 4 above)
Comment (KrzysztofMoch) [reply docs/pages/component/props.md:1]: Done ✅

File: docs/pages/component/events.md (hunk 3 above)
Comment (KrzysztofMoch) [reply docs/pages/component/events.md:3]: Done ✅

File: docs/pages/component/_meta.json (hunk 2 above)
Comment (KrzysztofMoch) [reply docs/pages/component/_meta.json]: Done ✅

**Comment by freeboub** [comment]
> OnAudioFocusChanged not documented (but it is not your fault :) )

To be followed in: https://github.com/react-native-video/react-native-video/issues/3314
Don't hesitate to open other documentation ticket if you want !

**Review by freeboub — APPROVED** [review]

**Pull request merged**